### DISCUSSION TITLE
iOS：PAGFile设置setDuration没有生效

### DISCUSSION THREAD

**Discussion by weidongjiang** [Tencent/libpag#1677] [Q&A] [answered]
版本：4.2.55
现象：在资源文件中有videos参数时，PAGFile设置setDuration没有生效。如下图。
![image](https://github.com/Tencent/libpag/assets/17646267/31fcd814-ec25-4307-9b0f-5d9ab946aae7)

问：
上面情况是一个bug，还是我们这个情况下就不支持设置setDuration。如果是一个bug，请问那个版本会修复。

**Comment by kevingpqi123** [comment, marked as answer]
可以仔细阅读下这篇文章：https://pag.art/docs/time-stretch.html
如果实现的过程中不符合预期，附上实现的工程，现在提供的信息有限，应该是你实现或理解的问题
none：不拉伸， 只有当设置的 duration 时长大于 pag 文件时长时生效，表现为多余的时间渲染的是最后一帧
scale：线性拉伸，当设置的 duration 和 pag 文件时长不一致时，拉取的区域为拉伸区间， 只有该模式有拉伸区间
repeat:重复，当设置的 duration 时长大于 pag 文件时长时生效
repeatinverted：倒序重复（播放到结尾后倒序播放， 当设置的 duration 时长大于 pag 文件时长时生效
